Add initial Knowledge Base post type handling

philcable opened the pull request:
* Registers the `knowledge_base` post type.
* Introduces template files for `knowledge_base` posts.
  * These can likely be stripped down quite a bit more.
* Introduces a `comments.php` template.
* Adds one CSS rule for specifically for Knowledge Base posts.
  * This can be incorporated into a properly named stylesheet later.

philcable committed:
Register a Knowledge Base post type

philcable committed:
Add template files for the Knowledge Base post type

philcable committed:
Add a comments template and call in the Knowledge Base post template

philcable committed:
Display modified date instead of publish date on Knowledge Base posts

Based on the project sheet: "The template displaying content for the
knowledge base post type should display related knowledge base posts,
user comments, and the *last updated date*."

philcable committed:
Remove "Published on " pseudo content for Knowledge Base posts

jeremyfelt commented on articles/post-knowledge_base.php:
```diff
@@ -0,0 +1,187 @@
+		}
+
+		// If a user has filled out their description and this is a multi-author blog, show a bio on their entries.
+		if ( is_singular() && get_the_author_meta( 'description' ) && is_multi_author() ) : ?>
```
Do we need author info on knowledge base posts? It may be okay just to remove this entire section.

philcable replied:
I agree with all of these suggestions! 🙂

jeremyfelt commented on articles/post-knowledge_base.php:
```diff
@@ -0,0 +1,187 @@
+		</hgroup>
+
+		<?php
+		if ( is_singular() && in_array( $post_share_placement, array( 'top', 'both' ), true ) ) {
```
I'd vote for removing the post sharing tools.

jeremyfelt commented on articles/post-knowledge_base.php:
```diff
@@ -0,0 +1,187 @@
+		}
+
+		// Display University locations attached to the post.
+		if ( taxonomy_exists( 'wsuwp_university_location' ) && has_term( '', 'wsuwp_university_location' ) ) {
```
I'd also support removing university locations (and university categories above) unless there's a real need to support them in the knowledge base.

jeremyfelt reviewed (approved):
Looks good - I left a couple optional notes about stuff that I thought could be removed from the template.

philcable committed:
Remove unnecessary output from the knowledge base post template

The share tools, University taxonomies, and author info aren't
particularly relevant to the post type.
